Capstone/Assets/Scripts/StateMachine/EnemyBehaviors/Actions/CanHit.cs

```diff
@@ -0,0 +1,32 @@
+using System.Collections;
+using System.Collections.Generic;
+using BehaviorDesigner.Runtime.Tasks;
+using UnityEngine;
+
+namespace Bladesmiths.Capstone
+{
+    public class Canhit : Action
+    {
+        public Canhit()
+        {
+
+        }
+
+        public override void OnStart()
+        {
+            base.OnStart();
+            GetComponent<Enemy>().CanHit = false;
+        }
+
+        public override TaskStatus OnUpdate()
```

Comment:
Can we get rid of this and OnEnd? Do we need them if all they're doing is calling base?